src/Sarif/Baseline/ResultMatching/DataStructures/MatchedResults.cs

```diff
@@ -46,7 +46,7 @@ public Result CalculateBasedlinedResult(DictionaryMergeBehavior propertyBagMerge
                 // Baseline result and current result have been matched => existing.
                 result = ConstructExistingResult(resultMatchingProperties, out originalResultMatchingProperties);
             }
-            else if (PreviousResult == null && CurrentResult != null)
+            else if (PreviousResult == null && CurrentResult != null && CurrentResult.Result.BaselineState != BaselineState.Absent)
```

Comment:
Do you need this after you added the code to only construct ExtractedResults for the ones which aren't absent?

Reply:
Sorry, this was where I was going to make the fix originally. I'll remove this change.

---
In reply to: [397397692](https://github.com/microsoft/sarif-sdk/pull/1826#discussion_r397397692) [](ancestors = 397397692)